connectors/sync_job_runner.py

```diff
@@ -189,6 +190,25 @@ async def _execute_access_control_sync_job(self, job_type, bulk_options):
             options=bulk_options,
         )
 
+    def _timestamp_optimization_enabled(self, job_type, data_provider):
+        """
+        Check if timestamp optimization is enabled for the current data source.
+        Timestamp optimization can be enabled only for incremental jobs.
+        """
+
+        # The job type is not incremental, so we can't use timestamp optimization
+        if job_type != JobType.INCREMENTAL:
+            return False
+
+        # Data provider has to have the method get_docs_incrementally (inherent from BaseDataSource or implemented in a subclass)
+        if not hasattr(
+            data_provider, "get_docs_incrementally"
+        ):
+            return False
+
+        # make sure that the data provider is not overriding the default implementation of get_docs_incrementally
+        return data_provider.get_docs_incrementally.__func__ is BaseDataSource.get_docs_incrementally
```

Comment:
You can simply check `data_provider.__class__.incremental_sync_enabled`

Reply:
I'm not sure if that helps. In that line, I'm checking if `get_docs_incrementally` is implemented in the data sources (connector) class. If a connector has `incremental_sync_enabled == True` and doesn't have a custom `get_docs_incrementally` then we can run a full sync with timestamp-based optimization.

Reply:
> If a connector has `incremental_sync_enabled == True` and doesn't have a custom `get_docs_incrementally` then we can run a full sync with timestamp-based optimization.

The variable `incremental_sync_enabled` is to indicate that the data source has incremental sync implemented.

Reply:
> The variable incremental_sync_enabled is to indicate that the data source has incremental sync implemented.

Unfortunately, even if this variable is True it doesn't mean that the connector has `get_docs_incrementally` implemented. 

Also, I need to understand what method I need to use at the framework level and there will be connectors with `incremental_sync_enabled==True` but without `get_docs_incrementally` implementation.